include/nlohmann/json.hpp

```diff
@@ -4265,6 +4250,23 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     basic_json* m_parent = nullptr;
 #endif
 
+#if JSON_DIAGNOSTIC_POSITIONS
```

Comment:
This is still boggling my mind

Reply:
the issue? yeah i have no idea, i'm just going with it now

Reply:
it inherits the private from the above fields, so its going to be private in its production usage anyways. No idea what Clang is doing